server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -128,6 +129,11 @@ private void validateParent(Mailbox mailbox, MailboxUpdateRequest updateRequest,
         }
     }
 
+    private boolean isParentIdInRequest(MailboxUpdateRequest updateRequest) {
+        return updateRequest.getParentId() != null
```

Comment:
By the way it seems awkward to me that updateRequest.getParentId() could be null.
 Maybe you can set it by default to Optional.empty() rathe than null ?